recipes/qpid-proton/all/conanfile.py

```diff
@@ -0,0 +1,144 @@
+        return "14"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("openssl/[>=1.1 <4]")
+        self.requires("cyrus-sasl/2.1.28")
+        if self.options.with_opentelemetry:
+            self.requires("opentelemetry-cpp/1.14.2")
+        if self.settings.os == "Macos":
+            self.requires("libuv/1.49.2")
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.16 <4]")
+        self.tool_requires("cpython/[>=3.9]")
+        if not self.conf.get("tools.gnu:pkg_config", check_type=str):
+            self.tool_requires("pkgconf/[>=1.7 <3]")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], filename=f"{self.version}.tar.gz", strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+
+        tc.variables["BUILD_PYTHON"] = False
+        tc.variables["BUILD_RUBY"] = False
+        tc.variables["BUILD_GO"] = False
+
+        tc.variables["ENABLE_WARNING_ERROR"] = False
+        tc.variables["ENABLE_UNDEFINED_ERROR"] = False
+        tc.variables["ENABLE_LINKTIME_OPTIMIZATION"] = False
+        tc.variables["ENABLE_HIDE_UNEXPORTED_SYMBOLS"] = False
+        tc.variables["ENABLE_FUZZ_TESTING"] = False
+        tc.variables["ENABLE_BENCHMARKS"] = False
+
+        tc.variables["BUILD_STATIC_LIBS"] = not bool(self.options["shared"])
+
+        tc.variables["BUILD_TESTING"] = False
+        tc.variables["BUILD_TOOLS"] = bool(self.options["tools"])
+        tc.variables["BUILD_EXAMPLES"] = False
+
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.set_property("libuv", "cmake_file_name", "Libuv")
+        deps.set_property("libuv", "cmake_target_name", "Libuv::Libuv")
+        deps.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, pattern="LICENSE.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(self, pattern="NOTICE.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+
+    @property
+    def _module_subfolder(self):
+        return os.path.join("lib", "cmake")
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "QpidProton")
```

Comment:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "ProtonCpp")
        self.cpp_info.set_property("pkg_config_name", "libqpid-proton-cpp")
```
The Correct CMake config file generate is Proton: https://github.com/apache/qpid-proton/blob/0.40.0/c/CMakeLists.txt#L688

Then, there is a separated one for C++: https://github.com/apache/qpid-proton/blob/0.40.0/cpp/CMakeLists.txt#L230 


The expected generate .pc file is declared here: https://github.com/apache/qpid-proton/blob/0.40.0/cpp/CMakeLists.txt#L42